lib/factory.js

```diff
@@ -27,7 +27,17 @@ Factory.prototype.createType = function(descriptor) {
   // initialize default values
   forEach(descriptor.properties, function(p) {
     if (!p.isMany && p.default !== undefined) {
-      prototype[p.name] = p.default;
+      var name;
+
+      if (p.ns.prefix === descriptor.ns.prefix) {
+
+        // use local name if property already in local ns
+        name = p.name;
```

Comment:
Interesting question here: What I'd expect is to have local name _and_ ns name attribute present.

I.e. I can always access the value via `foo["a:id"]` regardless if it has a local name set or not.

Likely this is only possible to accomplish via proxies, am I right about that?

Reply:
> Interesting question here: What I'd expect is to have local name and ns name attribute present.

They are present, but ...

> I.e. I can always access the value via foo["a:id"] regardless if it has a local name set or not. On the other hand I'd like to have a#id always available as the shorthand, if it is the only id property defined for an element.
>
> Likely this is only possible to accomplish via proxies, am I right about that?

For `isMany`-properties this is possible without a proxy, as we reference an object. But fundamental types (i.e. `BUILTINS`) are only accessible through `get()` when referenced with their `localName`. `a#id` on the other hand is always available through `a.id` (and `a.get('id')`of course), as long it was previously initialized (see my remarks below) - so yes, you're right.

IMO this is a compromise that we should accept. As you say, the alternative would be the introduction of proxy objects.

I hope we understand each other (please don't hesitate to keep raising questions). The test cases should hold the truth to specify desired behaviour. So we need to make sure to get them right and fully cover this topic as well as update the documentation before merging.

Btw, the current implementation also fails to access `foo['a:id']` if not initialized during creation or has a default value or was previously accessed with `get()`. I've not tested this out yet, but this seems to be the case for all properties except `foo.id`. I.e. you need to use `get()` first before `foo['a:id']` is mapped into the object.

This is somehow inconsistent and it might be a nice improvement to map property instances early (i.e. during type creation) even if they are not explicitly initialized or have a default value. I'd more like to get `undefined` instead of an unexpected exception. If you like, I'll open another issue for this.

One question in this context: should it be possible to define a default value for `isMany`-properties (e.g. `"default": [ 1, 2 ]`)? I tried this, received no exception but the value was not applied. Maybe I miss it, but there seems to be no documentation or spec for this. Should I address this in a seperate issue or am I just wrong here?